src/System.ComponentModel.TypeConverter/tests/AttributeCollectionTests.cs

```diff
@@ -90,28 +87,88 @@ public void ContainsTest(int count)
             }
         }
 
+        [Fact]
+        public void Contains_Attributes_ReturnsExpected()
+        {
+            Attribute[] attributes = GetAttributes().Take(5).ToArray();
+            var attributeCollection = new AttributeCollection(attributes);
+
+            Assert.True(attributeCollection.Contains(attributes));
+            Assert.True(attributeCollection.Contains(new Attribute[0]));
+            Assert.True(attributeCollection.Contains((Attribute[])null));
+            Assert.False(attributeCollection.Contains(new Attribute[] { new ReadOnlyAttribute(true) }));
+        }
+
+        [Theory]
         [InlineData(20)]
         [InlineData(1000)]
         [InlineData(1)]
         [InlineData(0)]
-        [Theory]
-        public void CountTests(int count)
+        public void Count_Get_ReturnsExpected(int count)
         {
-            var attributes = GetAttributes().Take(count).ToArray();
+            Attribute[] attributes = GetAttributes().Take(count).ToArray();
             var attributeCollection = new AttributeCollection(attributes);
 
             Assert.Equal(count, attributeCollection.Count);
             Assert.Equal(count, ((ICollection)attributeCollection).Count);
         }
 
+        [Fact]
+        public void FromExisting_NullExisting_ThrowsArgumentNullException()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("existing", () => AttributeCollection.FromExisting(null, new Attribute[0]));
+        }
+
+        [Fact]
+        public void FromExisting_NullAttributeInNewAttributes_ThrowsArgumentNullException()
+        {
+            Attribute[] existingAttributes = GetAttributes().Take(2).ToArray();
```

Comment:
What's the reason for the Take(2) vs Take(5) vs ...?

Reply:
Absolutely none - just shorter for the test data. The reason we don't just call `ToArray` is because the data is infinite - in a while true loop yielding data